### ISSUE TITLE
 Error starting userland proxy: mkdir /port/tcp:0.0.0.0:80:tcp:172.17.0.2:5000: input/output error.

### ISSUE THREAD

**Issue by signsmile** [docker/for-win#573]
### Information

  - Diagnostic ID from "Diagnose & Feedback" in the menu.
  - a reproducible case if this is a bug, Dockerfiles FTW
  - page URL if this is a docs issue or the name of a man page
  - host distribution and version (Windows version, build number, etc)

### Steps to reproduce the behavior

  1. ...
  2. ...

**Comment by friism** [comment]
Can you please add a diagnostic id and provide more details?

**Comment by Magdaflowne** [comment]
Hello

ID : 9F12A71E-4AE6-47EC-8210-1ED7FD3BEFA3/2017-03-22_12-10-00

Using docker compose with "Ports" enable to access web server in localhost.

Example :
ERROR: for mssql  Cannot start service mssql: driver failed programming external connectivity on endpoint serveurMsSQL (6ddab67975bf1feeae04902befbe6caf557e782859cfdf4a76e0a4fa47a251d2): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:1443
:tcp:172.18.0.3:1433: input/output error

**Comment by pf-curtis-mitchell** [comment]
Same error message for me.  Should I open a new ticket?
Just in case it's helpful here:
ID: 92AFBA65-1F89-4C80-8352-FE13334E71DC/2017-03-22_10-18-19
Version 17.03.0-ce-win1 (10296)
Channel: stable
e5a07a1
OS: Windows 10 Enterprise 10.0.14393 Build 14393

While doing further tests for this ticket, I made the following observation: 
The failing command was a simple `docker run` command that has a port binding with the full ip.  

For example
```
docker run -p 192.168.33.33:22:22 ...
```

The IP was incorrect. Once I made the correction, the command worked as expected.  

I hope that helps.

**Comment by friism** [comment]
@Magdaflowne do you have the docker-compose file that you're trying to run?

@pf-curtis-mitchell no reason to open a new ticket, and happy to hear you got it working.

**Comment by WrRaThY** [comment]
I also get this problem
OS: Windows 10 Enterprise
virtualization: native hyper-v

does not matter if I use powershell or ubuntu shell, same result

all you have to do to reproduce the issue is:
`docker run -p 8080:8080 hello-world`
or any other port for that matter

**Comment by apupier** [comment]
I also get this issue on an image which was working before.
I also reproduce with the simple "docker run -p 8080:8080 hello-world"

my version of Docker is 1.13.1 (10072)
I'm on Windows 10

I made no update of Windows, neither Docker between the working/failing state.

**Comment by WrRaThY** [comment]
what do you mean by "docker tooling" ?
my Docker version is 1.12.3, build 6b644ec (stable)
and if I check for updates I get the info, that its the latest one

**Comment by Magdaflowne** [comment]
Update in my post 

> Desactiving "Experimantal Features" (checked by default) solve my issue

**Comment by webdesignberlin** [comment]
Same Problem with Docker version 17.03.1-ce, build c6d412e
Windows 10 Enterprise v1607 Build 14393.953
virtualization: native hyper-v

**Comment by ajhodges** [comment]
Same issue...

ID: FFAB1CCB-ED9B-4799-A2F7-84B7501121FC/2017-04-06_11-01-38
Windows 10 Version 1607 Build 14393.726, using Hyper-V

Using this docker image: puckel/docker-airflow
https://github.com/puckel/docker-airflow

As @Magdaflowne said, unchecking "Experimental features" under "Daemon" seems to have fixed this issue.

**Comment by mswaminath** [comment]
Stop all the running containers (especially redis) and stop & `restart` the Docker on your machine.

**Comment by raghavmac** [comment]
1) Stop all the running containers `docker stop $(docker ps -a -q)` then
2) Stop the Docker on your machine & `restart` it.

**Comment by ikrysenko-selectica** [comment]
solution by @raghavmac works for me, thanks!

**Comment by ahemsath** [comment]
Windows Powershell equivalent of @raghavmac's one-liner to stop all running containers:
`docker ps -a -q | ForEach { docker stop $_ }`

**Comment by scopev24** [comment]
Hi!

Unfortunately we are currently facing the same errors.

We are running a freshly installed Windows Server 2016 where our developers work using RDP.

There are just two containers at the moment: apache-dev (based on centos:7) and redis-dev.

The redis container is started as follows:

> docker run --rm -v c:/apps/redis/redis.log:/var/log/redis.log -v c:/apps/redis/redis.conf:/usr/local/etc/redis/redis.conf:ro -itd -p 6379:6379 --name redis-dev redis:3 redis-server /usr/local/etc/redis/redis.conf

Both work as expected, but are not heavily used, since we're just evaluating docker in our new environment.
After a few hours (mostly over night) - I return to the server (which has not been used meanwhile) just to notice, that there seems to be something wrong - e.g. the php application running on the apache container complaining that redis is not reachable, or that it is not able to access our Oracle Database (which does not run in a containerized environment). Apache itself was reachable through the local browser at that time.

I then stop both containers.

If I try to start the containers again, I get something like this:

> C:\apps\redis>start_redis.bat
b33f8ccf8121de2e8e12e00e823ec891f40a104c8576a46d83e9b22a9de4ed9c
docker: Error response from daemon: driver failed programming external connectivity on endpoint redis-dev (b36dc0f21c4f40665b38dac87e5a4eb7482c72cd8c9430fadcecac1224be56b0): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:6379:tcp:172.17.0.3:6379: input/output error.

The batch file just contains the commmand from above.
If I use "Restart docker" on the "Reset" tab in the docker gui, I am again able to start both containers.

This issue hit me various times. Another example: I spawned a bash inside the apache container to test a few things but noticed, there was no internet connectivity. So I stopped the container and got the very same error on starting it. Restarted docker, started the container - spawned a bash, everything ok again.

It always seems like network connectivity gets lost after some time - and whatever happens prevents the containers to start again, after they have been stopped.

I uploaded a diagnostic, hoping that it helps:
0399F110-0301-424D-B0F7-3CC0E0C7C8CC/2017-08-22_09-16-15

I'm happy to provide more details - just don't know where to start my search...

Thanks!

Nico

**Comment by 22samurai** [comment]
Deactivating experimental features also worked for me

**Comment by oskaremil** [comment]
Deactivating experimental features also worked for me

**Comment by endlos** [comment]
Got this error today with no apparent  reason after working well always, and deactivating experimental features solved it.

**Comment by rmolinag** [comment]
+1 to fix by disabling experimental features, it would be nice to know which feature was causing it.

**Comment by Jamby93** [comment]
Anyone that has 'solved' disabling experimental features (that needs a reboot of the VM that we already know can fix the problem) has never seen it happen again for a month or more?
We continue experimenting this issue after some days or weeks of uptime, the only solution so far seems to reboot everything, disabling experimental features did nothing. 
This is the main issue that prevent us to going in production with Docker for Windows, we can't reboot production's vm every other week.

**Comment by daniel-gwilt-software** [comment]
Disabling "Experimental Features" in the Settings/Daemon menu, restarting docker and then stopping all images worked for me.

![image](https://user-images.githubusercontent.com/7307150/34236256-7976978a-e5ab-11e7-864e-69fca83f5f04.png)

**Comment by Jamby93** [comment]
@daniel-gwilt-software Worked for how long? More than a month of uptime?

**Comment by adampl** [comment]
The same issue. Each time I open my Windows 10 laptop (boot the system) and the Docker service starts, I'm unable to bring my Docker Compose project up, because:

```
Creating code_application_1 ... error

ERROR: for code_application_1  Cannot start service application: driver failed programming external connectivity on endpoint code_application_1 (bd7f89ef4c1b0c6cbb4eb82ba552e5ccbf87f168ad81b8f8656bdc6443c2ef79): Error starting userland proxy: mkdir /port/tcp:0.0.0.
0:8888:tcp:172.18.0.4:8000: input/output error

ERROR: for application  Cannot start service application: driver failed programming external connectivity on endpoint code_application_1 (bd7f89ef4c1b0c6cbb4eb82ba552e5ccbf87f168ad81b8f8656bdc6443c2ef79): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:8888:
tcp:172.18.0.4:8000: input/output error
ERROR: Encountered errors while bringing up the project.
```

Here is the diagnostic ID: `65B03A6A-4D0B-4364-A84C-41DDE0B4AB0F/2017-12-29_14-27-34`

After I stop the service (quit docker) and then start it again, everything works fine.

Here is my compose file:
```
version: "3.4"

services:
  data:
    image: postgres
    env_file:
      - vars.env
  messages:
    image: rabbitmq
    env_file:
      - vars.env
  application:
    build: .
    env_file:
      - vars.env
    command: gunicorn --bind=0.0.0.0:8000 --workers=4 proj.wsgi
    volumes:
      - .:/code
    ports:
      - "8888:8000"
    depends_on:
      - data
      - messages
```

**Comment by DGolubets** [comment]
Yeah, as @rickvanbodegraven said - it has nothing to do with experimental features, it is just restarting  Docker that helps.

I have Win 10 with docker Postgres container for development bound to my local port. Every time I restart PC - i have to recreate container, otherwise it won't work. That's annoying.

**Comment by piers7** [comment]
I had this issue today with Docker for Windows 10 (x64, Community) Version 17.09.1-ce-win42 (14687). As other have said, restarting the daemon fixed it (I still have experimental features enabled).

**Comment by philandrews-ixxus** [comment]
+1 for this issue.  Previously I had the cold boot startup error where docker would popup a timeout when I first booted my PC (Windows 10 Pro version 1709) but that went away and docker seemed to be working fine until I had to start interacting with containers then I would get this userland proxy issue requiring a VM reboot.

**Comment by daniel-gwilt-software** [comment]
@Jamby93 I've only been using my current docker set up for a couple weeks, but it is currently working.

@rickvanbodegraven You may be correct about that.  I tried starting my image today, got the proxy error, simply restarted docker and everything worked again.

**Comment by petegore** [comment]
Got the same issue with mysql image today. Worked fine yesterday.
Solved it by setting "experimental features" off.

**Comment by rickvanbodegraven** [comment]
To expand on this issue, I've noticed that it happens after I use OpenVPN to connect to our office network (which does all sorts of stuff to my routing and network interfaces). It also happens when I use my laptop through its USB-C docking station, which conveniently adds a new network interface whenever it is plugged in. 

I am starting to suspect that when the Docker VM is booted, it gets attached to a specific network interface, and whenever that network interface no longer exists, this problem pops up. Simply restarting it (for example by clicking the `Apply` button) will cause the VM to attach to a different network interface, and thus solving the issue for however long that interface exists.

**Comment by garyritchie** [comment]
Company VPN suspected here, too. It wasn't running at the time of the error as I had disconnected about an hour ago. Restarting docker didn't do it for me -- I went straight for reset. Not ideal.

**Comment by blackholegalaxy** [comment]
More on that issue. Happens to me on a computer with no VPN at all (only bitdefender Security Suite, firewall ok to let docker do its job). At windows start: the error occurs. Demon restart: everything is ok.

**Comment by Lufan** [comment]
This issue happen a lot of time for me (win 10 pro, Docker version 17.09.1-ce, build 19e2cf6). It is not dependent of status of  the "Experimental features". And I need each time to restart the docker daemon to continue working.

**Comment by kents1** [comment]
I have the same issue; after shutting down Windows and restarting, then trying to start via docker-compose.  I can confirm that simply restarting the docker daemon fixes the issue until the next reboot.

**Comment by boukeversteegh** [comment]
@rickvanbodegraven I had the same error, but after restarting i can "docker-compose up" without errors, but my containers are not reachable using localhost.
I'm using a USB3-dock for ethernet as well. Maybe it causes problems.. :-/

**Comment by dankempster** [comment]
Some more info, maybe this will help?

The  issue happens daily for 3 of 4 developers within my team. They're all running Windows 10 Pro, on a corporate domain (I'm using a Mac and don't have this issue).

**Diagnostic Ids:**
Problematic machines:
 - 46586E4A-835C-4D36-B271-CBC401C9614A/2018-01-11_09-30-17
 - 49915569-1176-4094-88E9-94060114F136/2018-01-11_09-30-23
 - AD2AC47A-53E0-47AC-87E6-0C57166E93C0/2018-01-11_09-30-34

OK machine:
 - 6FB35428-155B-4EAF-95F3-539789086D26/2018-01-11_09-30-33

We use Docker[-compose] for our development environments. Unfortunately, I am unable to share the 
exact docker-compose files we use.

Heres the type of output they get:
> ERROR: for feature970stockdetailstopremiseslevel_db_1  Cannot start service db: driver failed programming external connectivity on endpoint feature970stockdetailstopremiseslevel_db_1 (0fc7f853b3f1de48c5710490925fa7008d9705474734ec41201c6f156c8caf1a): Error star
ting userland proxy: mkdir /port/tcp:0.0.0.0:3006:tcp:172.18.0.2:3306: input/output error

Also, one of my developers (ID: 49915569-1176-4094-88E9-94060114F136/2018-01-11_09-30-23)  experienced the localhost issue (reported by @boukeversteegh, above) for the first time today. Restarting docker (for a second time this morning) fixed the localhost issue.

As I have access to three machines with the issue and one without, I'm happy to aid with any diagnostic/investigation to get this issue identified & resolved.

**Comment by damacus** [comment]
@friism what debugging information do you need to get this investigated a little deeper? 

Thanks!

**Comment by pldin601** [comment]
I got same issue after my PC fall into sleep. Windows 10 Pro. Fixed by restarting Docker.

**Comment by Jamby93** [comment]
I would like to have a 'fixed by restarting' label so that we could close basically every issue in the IT world with that solution.

**Comment by xtreamwayz** [comment]
I'm having the same issue.

> Desactiving "Experimantal Features" (checked by default) solve my issue

This fixed it temporary because it restarts docker. When shutting down my laptop the same issue is back after starting it again.

I did some more investigation and I found out that since the Fall Creators Update windows has a new "hybrid shutdown" feature. What it does is storing the state of open applications and reopen those when you start your pc/laptop again. This is the default behavior when clicking "start -> power -> shutdown".

This behavior can't be switched off, but you can shutdown manually in your terminal as before:
`shutdown.exe /s /t 0` _(Adding `/hybrid` would enable the current behavior)_

If I shutdown with this command I don't have the mentioned issues  anymore.

According to the release notes of [insider-preview-build-17040](https://blogs.windows.com/windowsexperience/2017/11/16/announcing-windows-10-insider-preview-build-17040-pc/) this hybrid shutdown behavior can be disabled starting from 17040. 

> Based on your feedback, the feature to restore applications that have [registered for application restart](https://msdn.microsoft.com/en-us/library/windows/desktop/bb525423(v=vs.85).aspx) after you reboot or shutdown (through power options available on the Start Menu and various other locations) has been set to only occur for users that have enabled “Use my sign-in info to automatically finish setting my device after an update or restart” in the Privacy section under Sign-in Options Settings.

I'm guessing that somehow docker must detect that the device was restarted after a hybrid shutdown and clear some settings first.

**Comment by Zabanya** [comment]
@xtreamwayz Thanks, I'm using 1703, with business branch and 2 months delay, i got no update, with 0 day delay i got 1709 one, hopefully it does not create another problem while i am trying to fix one. if this working fine i'll delay for another year or so.

**Comment by ghayes7** [comment]
Restarting docker (Windows, Docker Settings --> Reset --> Restart Docker) fixed the issue for me.

**Comment by adampl** [comment]
@friism Do you still need some info?

**Comment by Diluka** [comment]
I have the same issue when I upgrade to the lastest version. Restart docker will fix it temporarily.
A diagnostic was uploaded with id: 11F87FC1-FCD5-4980-B8D1-E70519A3C413/2018-01-28_19-32-57

**Comment by saostad** [comment]
I am restarting docker every day! 
what is permanent solution?

**Comment by xtreamwayz** [comment]
@saostad Did you try this https://github.com/docker/for-win/issues/573#issuecomment-359162168 ?

**Comment by cmaslen** [comment]
As with many other comments I have the same issue from a cold reboot with a restart of Docker fixing the issue. comment above from @xtreamwayz  worked for me. The explanation regarding hybrid shutdwn sounds plausible to me so perhaps the Docker Win devs have something more concrete to go on.

**Comment by Jamby93** [comment]
Same situation of @saostad here: we are restarting Docker For Windows everyday or multiple times a day on Windows Server 2016, no shutdown, no network card issue, just some cointainers that "lose" public port forwarding after a while and no new container with port forwarding can be started, with the aforementioned error. It's basically impossibile to use in production given that the first clue of this problem already involves service interruption and daemon's restarting require manual intervention.

**Comment by socialknowledge** [comment]
I tried restarting Docker to no avail.  Finally I disabled experimental features and it worked.

**Comment by blackSourcez** [comment]
@daniel-gwilt-software Thanks, it's works for me.

**Comment by magic890** [comment]
Same problem, restarting Docker on Windows solved it.

**Comment by danbrellis** [comment]
I turned off Experimental Features and then turned it back on and the problem went away. Seems to suggest it is the restarting of docker instead of the experimental features itself.

**Comment by Symbianx** [comment]
I don't think the issue is related to docker itself, but the linux virtual machine it is running on.
Restarting docker solves the issue because it also restarts the VM.

The issue happens again when you restart your machine because Hyper-V does not restart the VM, instead it suspends the VM. Somehow this is causing the port to look like it is being used after restoring.

**Comment by BGaunitz** [comment]
@danbrellis Helped me as well, thx

**Comment by kexplo** [comment]
Deactivating experimental features also worked for me, thanks!

**Comment by rexthk** [comment]
I have the same issue. On windows 10, using mysql image.

**Comment by Jeern** [comment]
Hmm. I have the same problem. I guess it basicly means that docker for windows is unusable for production, but ok for development. A big issue for those who are dependent on windows servers. Not so much for others. I hope it will be fixed asap.

**Comment by adampl** [comment]
@Jeern for sure Windows is unusable not production, not to mention Docker for Windows ;)

**Comment by Jeern** [comment]
@adampl he he. :) Well it would pretty harsh to blame Windows for not working optimally with a Linux tech. Redis also works better on Linux although in that case the Windows version is actually pretty solid and ok for production. This issue here is  bad though, not so much for those of us using Linux servers with kubernetes in production, but I would imagine it Kills docker for some shops

**Comment by adampl** [comment]
@Jeern Behind the scenes, Doker for Windows uses a virtual machine with Linux and docker on it, so I'd guess you can overcome this issue by just doing it yourself.

Personally, I was using Docker for Windows for local development, but recently I switched to a Ubuntu desktop and now I'm really satisfied with the performance and stability, so I recommend it to anyone who's serious about using Docker with Linux images.

**Comment by Jeern** [comment]
@adampl I know and true. I do not have that option currently though.

**Comment by t-o-o-m** [comment]
@xtreamwayz lead me to a permanent "fix": after disabling fast startup (Power Options -> Choose what the power buttons do -> Change settings that are currently unavailable -> Turn on fast startup (recommended) -> uncheck) docker networking works as expected. No need to turn off “Use my sign-in info to automatically finish setting my device after an update or restart” in the Privacy section under Sign-in Options Settings. And no need to shut down with a specific command everytime. 
No fast startup anymore, though; I can live with that!

**Comment by Zabanya** [comment]
anyone tried using legacy restart / shutdown window?, which can be accessed by pressing alt + f4 on desktop.

**Comment by xtreamwayz** [comment]
> anyone tried using legacy restart / shutdown window?, which can be accessed by pressing alt + f4 on desktop.

Just tried, doesn't work.

Turn off fast startup does work. I guess I go with that for now. Thanx @t-o-o-m.

**Comment by t-o-o-m** [comment]
If Windows 10 doesn't offer the option for fast startup (many causes for that, e.g. turned off hibernation), turning off “Use my sign-in info to automatically finish setting my device after an update or restart” in the Privacy section under Sign-in Options Settings works as well, as already pointed out earlier.

**Comment by Kubitomakita** [comment]
Permanent fix suggested by @t-o-o-m works like a charm. I've been resetting Docker after every boot since months...

**Comment by ruifigueira** [comment]
I couldn't find any UI to turn off **“Use my sign-in info to automatically finish setting my device after an update or restart”** as @t-o-o-m pointed out, so I went directly to regedit and created a Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System\DisableAutomaticRestartSignOn key (DWORD type) with value 1 and it seems to be working fine now.

More details on that property can be found in Winlogon Automatic Restart Sign-On (ARSO).

**Comment by wilsonferro** [comment]
![image](https://user-images.githubusercontent.com/2805584/38315043-fccab8e2-37fd-11e8-879d-ba54c0f09237.png)

**Comment by crramirez** [comment]
Just restarting docker worked for me

**Comment by tnovau** [comment]
Desactiving "Experimantal Features" solve my issue

**Comment by DGolubets** [comment]
Guys, please stop saying"Experimental Features helped me".  It's either: a) you had a different problem or b) you didn't test it. So it's only confusing everyone else coming to this thread.

The only **permanent** workaround is the one suggested by @xtreamwayz and @t-o-o-m: turning off the fast startup. 

P. s. 
Unsubbing from this thread, cos I have no hope of an official answer anymore 😞

**Comment by crramirez** [comment]
With a little discipline, instead of disabling fast startup, just stop the containers or exit docker for windows before shutting down the PC. It works for me.

**Comment by 99linesofcode** [comment]
Just wanted to confirm that disabling Windows Fast Startup offers a workaround for this issue. Hopefully this is going to receive some love by someone who is bored working on 'high priority' issues.

For those of you who struggle to find the option to enable / disable fast startup, it is there but a tad hidden. No need to regedit 😉

1. Go to _Control Panel_
1. Find _Power Options_
1. Click on _Choose what the power button does_ in the sidebar on the left.
1. Click on _Change settings that are currently unavailable_ to allow checking / unchecking options under shutdown settings
1. Toggle _Turn on fast startup_

**Comment by kodaniel** [comment]
Disabling the fast startup is not a 100% solution, only works if you restart the OS, but don't if you just log out and in. But most of the cases it's enough.

**Comment by ruifigueira** [comment]
@99linesofcode, thanks a lot, It worked perfectly! My previous "solution" with `regedit` did not work, it only seemed to work for a couple of times, but I ended up having the same issue once again.

Here's a screenshot with the steps once you click on `Choose what the power button does`:

![image](https://user-images.githubusercontent.com/1374559/39571814-a9276338-4ec4-11e8-89b8-140409ff80ec.png)

**Comment by alessiofiore** [comment]
Solution by @raghavmac works for me too. Many thanks!

**Comment by ProggerPete** [comment]
Having to restart docker to fix this is tedious.

**Comment by fakocher** [comment]
I had the same issue and restarting Docker solved it.

**Comment by dalibor983** [comment]
What I can notice docker containers working well.
You can execute commands in running containers with ```docker exec -it <running-container> /bin/bash``` but after restarting windows hyper-v networking is totally messed up.

Currently only docker restart is a fix for this problem.

**Comment by dalibor983** [comment]
I notice in hyper-v settings for MobyLinuxVM vm that in **Automatic Stop Action** is selected **Shut down the guest operating system**. But there is a notice **The integration service that controls shutting down the guest operating system must be installed and enabled on the virtual machine.**

Dose MobyLinuxVM have this **integration service** and does it works?

Maybe the MobyLinuxVM does not shout down itself when host machine is shut down.

**Comment by Matts** [comment]
As @DGolubets stated, the reason turning experimental features off works. Is because it restarts the docker daemon. Indeed after disabling the login option, it no longer occurs for me.

**Comment by agonza1** [comment]
In my case switching to enable experimental features fixed the issue. So it seems to be the Docker reboot what is fixing this issue.

**Comment by samsonkolge** [comment]
Deactivating "Experimental Features" and hitting "Apply" did the trick for me.
To verify:
```
$ docker version -f '{{.Server.Experimental}}'
false

```

**Comment by saostad** [comment]
I am restarting my docker everyday for 2 months now!
is there any hope to fix?

**Comment by Jamby93** [comment]
@saostad After months in your identical situation (actually I ended up rebooting the VM multiple times a day depending on the number of services deployed and active during the day) we dropped Docker for Windows and simply switched to native boot2docker in a VM for Linux container and a Windows 1803 VM for Windows containers both is the same swarm. We also tried LCOW but it seems too early for using it in production (just as example, LCOW containers will see just 2GB of RAM). 
We're using Vagrant to deploy always-working VM but docker-machine should do the work as well, depending on your infrastructure.
If "reboot the software" is a solution, so should be "uninstall the software and use something else", isn't it? :)

**Comment by michaelpapesch** [comment]
Really guys? Are you joking? Stop posting in this thread, if you have not READ IT.

In this thread the workaround for the problem was mentioned at least 5 times, with screenshots and everything, and it gets buried by stupid people just coming here to post "Disabling experimental features helped me!" or "Enabling experimental features helped me!".
It does not, stop posting it!

READ THE THREAD, its only 5 minutes until you get to the solution.
Better than trying to fix your problem for hours now, isn't it?
5 minutes of careful reading vs. 3 hours of brainless button clicking.
Can you do it?

**Comment by SaadQamar01** [comment]
In my case, I restarted the docker app so it was working fine :)

**Comment by whitecolor** [comment]
Sometimes happening to me, sometimes not:
```
ERROR: for dnsdock  Cannot start service dnsdock: driver failed programming external connectivity on endpoint dnsdock (0ead68149bc2bb71a64cf1bfa76e9cdbfc2ef3702971e148987b544316f3988a): Error starting userland proxy: mkdir /port/udp:0.0.0.0:54:udp:172.18.0.3:53: input/output error
```
Hope disabling experimental features will make this stable.

**Comment by interpeix** [comment]
In my case to reproduce the error is quite easy, just logout...all the running containers die...after login again the container seems to be working(but it's not) stop and start the container and the error pops up.

It seems that the mobylinux die.

**Comment by marcin-slezak** [comment]
Hi, the same issue - docker restart helps

**Docker:**: Version 17.12.0-ce-win47 (15139),  Channel: stable, 9c692cd

**Docker diagnostic id**: B5165DDB-C80F-45AB-A209-3D112B3FDE45/2018-06-27_08-39-01

**OS**: Microsoft Windows 10 Pro, Version 10.0.17134, Build 17134

**docker-compose.yml**
`version: '3.3'
services:
 db:
  image: postgres
  restart: none
  environment:
   POSTGRES_PASSWORD: example
  ports:
   - "5432:5432"
 adminer:
  image: adminer
  restart: none 
  ports: 
    - "8080:8080"`

Error:
> Error response from daemon: driver failed programming external connectivity on endpoint some-postgres (9349410aa5233a43fc53d186a20a467c933e2a47b9a4b7722dc6f19c6ba002db): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:5432:tcp:172.17.0.2:5432: input/output error

**Comment by firdanirp** [comment]
I was resolve this problem with restart my docker, my OS is Windows 10. This problem occured because docker not stop correctly. We need to exit docker before shutdown our pc

**Comment by JordyLangen** [comment]
@rickvanbodegraven 's fix worked for me, and his possible explanation seems to align with my working environment.

**Comment by ajkret** [comment]
I am running Win 10 Enterprise on a corporate domain, build 1803, and docker 18.03.1-ce-win64 (17438). I have a docker swarm compose file (can't show it due to Non-disclosure agreement) and a SONAR image. I have to restart Docker for the SONAR container run again after every reboot, even after disabling HYBRID shutdown. The problem is solved only if I restart docker AND run SONAR in a new container.

```
PS C:\Users\a.kretschmer> docker start sonar
Error response from daemon: driver failed programming external connectivity on endpoint sonar (d0b33c01b2fdef1112344edf4
2def9c2f80973da7cb64f64e343c9af2eebaf3f): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:9292:tcp:172.17.0.2:909
2: input/output error
Error: failed to start containers: sonar
```

**Comment by kleky** [comment]
I think it helps to bring containers down before restarting windows

**Comment by jeacott1** [comment]
had this today. had to restart docker a few times before it came good.

**Comment by kleky** [comment]
I had this many times before, but a simple restart got it working again. Yesterday was a bad one and I was out of action for 2 hours, but then one restart just got it working. a total mystery. and pain.

**Comment by leonbrag** [comment]
I just run into this issue on Win10. Restarting docker fixed it. Version:

**docker info**
```
Containers: 3
 Running: 2
 Paused: 0
 Stopped: 1
Images: 5
Server Version: 18.06.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: d64c661f1d51c48782c9cec8fda7604785f93587
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.93-linuxkit-aufs
Operating System: Docker for Windows
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.934GiB
Name: linuxkit-00155df86909
ID: WUXF:X2GQ:6GQ2:7NLR:CX7C:WGDY:DNRC:ZGL2:N5CX:247L:VMXW:ZSGH
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 30
 Goroutines: 62
 System Time: 2018-07-30T22:34:12.7098407Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by ProggerPete** [comment]
Amazing that a bug this bad still exists over a year after it was reported.

**Comment by vivaladan** [comment]
An issue for me too! Computer going to sleep caused it. Docker restart fixed it.

**Comment by aahmadts** [comment]
I had the same problem, just **restarted Docker** aaaaaaaand it worked

**Comment by xduseko** [comment]
This is so annoying. Every time I start docker I have to wait until it's ready and then restart it and wait again. Hope fix is coming soon :-)

**Comment by amitcsknit** [comment]
running this docker ps -a -q | ForEach { docker stop $_ }
& restarting docker works for me

**Comment by mario-bros** [comment]
Hi Guys, actually i manage to solve this issue on my windows by increasing docker swap memory which can be done with GUI, go to Setting -> Advanced like pic below : 



![capture](https://user-images.githubusercontent.com/8444528/45071404-432dee00-b100-11e8-9551-1d21e4c92bde.PNG)


then i increase swap memory from 1024MB to 2048MB, before i do this, docker ps -a -q | ForEach { docker stop $_ } didnt work for me.

**Comment by michaelpapesch** [comment]
@mario-bros 
no, you did not solve it, you only restarted docker with that.
restarting is atm the only thing that helps.

**Comment by tfink** [comment]
Same problem for me, when docker starts the first time on windows, it can't download images from the internet nor start something with a port usage:
ERROR: for foo_nginx_1  Cannot start service nginx: driver failed programming external connectivity on endpoint foo_nginx_1 (2502ff7db901058c000c12ee3096497744ec6fde93bab20eee026b3248016378): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:80:tcp:172.20.0.4:80: input/output error

Stopping and Restarting the demon solves this problem until the next boot.

**Comment by qoricode** [comment]
Yes - same problem here with latest version 18.06.1-ce-win73 (19507), I have to restart dfw to fix it.

**Comment by yuripallada** [comment]
Same problem here with latest version 18.06.1-ce-win73 (19507). Stopping and restarting doesn't actually solve this problem. 

I reinstalled Docker which fixed this problem.

I'm also having issues with mapping container ports to host ports using -p argument in the docker run command. It does run the container, but opening a browser on localhost using the host port doesn't do anything. (Windows x64 Pro)

**Comment by j-clamen** [comment]
here is a permanent workaround working for me (Windows 10 Pro, Docker CE). I do not need to restart docker every time. Compromise is your system time to start all docker services. Step 1/ remove Docker from program startup list and Step 2/ run Docker from a batch with a delay. 120 seconds works well I will try later to reduce the delay 

@echo off
CONSOLESTATE /hide
timeout /t 120 /nobreak > nul
Start "" "C:\Program Files\Docker\Docker\Docker for Windows.exe"
exit

**Comment by vardhmanandroid2015** [comment]
removing **experimental** tag from docker daemon "**which was set to True**" solved this issue for me.... Now I am able to make the docker container up with exposing container port to host post

**Comment by songpr** [comment]
same problem with Windows 10 Pro and Docker Community Edition
Version 18.06.1-ce-win73 (19507)
Channel: stable
b0f14f1

It's waste of time that I have to restart Docker everytime, I start my windows

**Comment by johnknoop** [comment]
I've had this exact problem for a long time. Since I'm hosting my local MySQL in Docker, it's a real hassle that I need to restart the daemon (and then also restart the container, but that might be unrelated?) every morning when I come to the office.

**Comment by vivaladan** [comment]
This is the most commented issue on docker for Windows. If it's still tagged as more information needed, then what information is needed. Is this being investigate or is it on ice. Is this something the community can help with?

**Comment by ChuckkNorris** [comment]
Encountering the same issue with "Experimental features" turned off. Each time I restart my computer, I need to manually restart docker.

Not sure if it's related, but I do have a network defined in my `docker-compose.yml`:
```
version: '3.4'

services:
  # Rabbit MQ Server
  rabbitmq:
    image: rabbitmq:3-management-alpine
    hostname: "rabbitmq"
    environment:
      RABBITMQ_ERLANG_COOKIE: "cookie"
      RABBITMQ_DEFAULT_USER: "user"
      RABBITMQ_DEFAULT_PASS: "pass"
      RABBITMQ_DEFAULT_VHOST: "/"
    ports:
      - "5672:5672"
      - "5671:5671"
      - "15672:15672"
    networks:
      - webnet
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:15672"]
      interval: 30s
      timeout: 10s
      retries: 5

  # Catalog Microservice
  catalog.data:
    image: microsoft/mssql-server-linux:2017-latest
    # user: $UID # "sa" if not set
    environment:
      ACCEPT_EULA: Y
      SA_PASSWORD: "MyPassword"
    ports:
      - "1433:1433"
    volumes:
      - mssqldata:/var/opt/mssql
    networks:
      - webnet

  catalog.api:
    restart: on-failure
    build:
      context: .
      dockerfile: Microservices/Catalog.Api/Dockerfile
    networks:
      - webnet
    ports:
      - "4001:80"
    depends_on:
      - catalog.data
      - rabbitmq
  
  # Basket Microservice
  basket.api:
    build:
      context: .
      dockerfile: ./Microservices/Basket.Api/Dockerfile
    networks:
      - webnet
    ports:
      - "4002:80"
    depends_on:
      - rabbitmq

volumes:
  mssqldata:

networks:
  webnet:
    driver: bridge
```

Using Docker CE, v18.06.1-ce-win73 (19507)

**Comment by jmartsch** [comment]
This also happens to me with Docker Version 18.06.1-ce-win73 (19507) on Windows 10 Professional Version 1803

**Comment by znmeb** [comment]
At this point I'm inclined to believe that there is no fix to be made - that it's a fundamental flaw in the way Windows shutdown works. Docker for Windows needs to determine when a Windows shutdown is happening and gracefully shut all the containers running in the Hyper-V Moby Linux VM, as well as assuring that the VM itself is gracefully shut down. My conjecture is that there's no mechanism available for this. I'd love for someone ***in a position of authority at Docker*** to prove me wrong.

**Comment by leonbrag** [comment]
https://docs.microsoft.com/en-us/windows/desktop/shutdown/shutting-down

Shutdown Notifications
Applications with a window and message queue receive shutdown notifications through the WM_QUERYENDSESSION<https://docs.microsoft.com/en-us/windows/desktop/shutdown/wm-queryendsession> and WM_ENDSESSION<https://docs.microsoft.com/en-us/windows/desktop/shutdown/wm-endsession> messages. These applications should return TRUE to indicate that they can be terminated. Applications should not block system shutdown unless it is absolutely necessary. Applications should perform any required cleanup while processing WM_ENDSESSION. Applications that have unsaved data could save the data to a temporary location and restore it the next time the application starts. It is recommended that applications save their data and state frequently; for example, automatically save data between save operations initiated by the user to reduce the amount of data to be saved at shutdown.
Console applications receive shutdown notifications in their handler routines. To register a console handler, use the SetConsoleCtrlHandler<https://msdn.microsoft.com/library/ms686016(v=VS.85).aspx> function.
Service applications receive shutdown notifications in their handler routines. To register a service control handler, use the RegisterServiceCtrlHandlerEx<https://msdn.microsoft.com/en-us/library/ms685058(v=VS.85).aspx> function.

**Comment by znmeb** [comment]
@leonbrag So are you saying it's possible for Docker for Windows to save container state on shutdown and restore it on startup?

**Comment by leonbrag** [comment]
All I am saying that per Windows documentation, applications are notified of the shutdown event.

**Comment by maheshrpm** [comment]
Got same issue on Win 10.
Restarting Docker solved the issue.

**Comment by systemsincode** [comment]
same eror, same fix (docker restart) for me using 18.06.1-ce-win73 (19507)

**Comment by Dzieniu2** [comment]
Same problem, looks like restarting docker solves the problem.

**Comment by WilliamStam** [comment]
+1 restarting docker works. 
inconvenient

**Comment by uwieske** [comment]
I am experiencing the same problem. As a workaround I have to restart Docker every time this error happens. Judging by the amount of reactions above the last months... I suppose that the solution to this problem is really difficult. Can ony one point me (us) to the solution, please?

**Comment by songyahao** [comment]
solution by @raghavmac works for me, thanks!

**Comment by ProggerPete** [comment]
> solution by @raghavmac works for me, thanks!

Restarting docker is not a solution.  It is a workaround.

After 18 months it seems pretty clear that docker on windows is not a priority.  My 'solution' was to abandon using it and run a VM instead.

**Comment by gmaf** [comment]
Had the same issue after updating to ```Version 2.0.0.0-win78 (28905)```.
Nothing above helped.
Only ```Reset to factory defaults``` helped:
![image](https://user-images.githubusercontent.com/12168520/48764482-36657300-ecc9-11e8-8991-89d4248484ce.png)

**Comment by znmeb** [comment]
I'm in the process of migrating to Fedora Atomic Host in Hyper-V. That way, at least I'll be able to troubleshoot things like this and open issues. I'll post a link to the repo when I get things going; it'll probably be a week or so till I get it going; I only use Windows now once or twice a week and I'm not a PowerShell expert.

**Comment by lucyandsandy** [comment]
restart dockers solved my problem

**Comment by Kitkevich** [comment]
Same error message for me

**Comment by Zaradar** [comment]
Same error - restart does the trick. Can anyone confirm if this flaw also affects Docker EE or is it only hitting CE?

**Comment by ajkumar01** [comment]
For me the issue went away, when i invoke 'docker run -p123:456 .... ' from a powershell that was run as 'Administrator'.

Version 2.0.0.0-win78 (28905)
Channel: stable
Build: c404a62
(docker engine version: 18.09.0)

**Comment by vetras** [comment]
Disable experimental features and restart docker on windows worked for me.

**Comment by GhitaCristian** [comment]
I had the experimental features turned off by default. Pressing 'Restart Docker Desktop...' in Reset section fixed it for me on win 10. 
![image](https://user-images.githubusercontent.com/15012763/49375173-7847dd80-f6fb-11e8-96dd-e23e2da7cdf1.png)

**Comment by codersk** [comment]
> 1. Stop all the running containers `docker stop $(docker ps -a -q)` then
> 2. Stop the Docker on your machine & `restart` it.

Worked for me

**Comment by tmcentee** [comment]
I know this has been stated many times in this thread, but everything regarding "restart docker" isn't a fix. 

They are workarounds to a real issue that is obviously affecting many people day-to-day. 

@friism is there anything else we can do to get this looked at?

**Comment by friism** [comment]
ping @carlfischer1

**Comment by elcamlost** [comment]
faced issue today. restart docker was helping.
But it's not a solution for everyday work. 

Please, fix it.

**Comment by matevarga** [comment]
Try setting `COMPOSE_PARALLEL_LIMIT=4` or something similar. I don't think this is a DfW-specific problem.

**Comment by coreyman** [comment]
Had this issue today. Restarted docker and started my container. It asked me to allow vpnkit through the firewall. I allows it through the firewall and all is working.

**Comment by NorthNick** [comment]
> I did some more investigation and I found out that since the Fall Creators Update windows has a new "hybrid shutdown" feature. What it does is storing the state of open applications and reopen those when you start your pc/laptop again. This is the default behavior when clicking "start -> power -> shutdown".
> 
> This behavior can't be switched off, but you can shutdown manually in your terminal as before:
> `shutdown.exe /s /t 0` _(Adding `/hybrid` would enable the current behavior)_
> 
> If I shutdown with this command I don't have the mentioned issues anymore.
> 
> According to the release notes of [insider-preview-build-17040](https://blogs.windows.com/windowsexperience/2017/11/16/announcing-windows-10-insider-preview-build-17040-pc/) this hybrid shutdown behavior can be disabled starting from 17040.
> 
> > Based on your feedback, the feature to restore applications that have [registered for application restart](https://msdn.microsoft.com/en-us/library/windows/desktop/bb525423(v=vs.85).aspx) after you reboot or shutdown (through power options available on the Start Menu and various other locations) has been set to only occur for users that have enabled “Use my sign-in info to automatically finish setting my device after an update or restart” in the Privacy section under Sign-in Options Settings.
> 
> I'm guessing that somehow docker must detect that the device was restarted after a hybrid shutdown and clear some settings first.

Turning off fast startup in Control Panel, Hardware and Sound, Power Options, System Settings solved the problem on my Win 10 laptop. Docker takes a lot longer to start after bootup now, but containers run properly, while one of them reliably needed a service restart before.

**Comment by muliar** [comment]
> 
> 
> Had the same issue after updating to `Version 2.0.0.0-win78 (28905)`.
> Nothing above helped.
> Only `Reset to factory defaults` helped:
> ![image](https://user-images.githubusercontent.com/12168520/48764482-36657300-ecc9-11e8-8991-89d4248484ce.png)

Your solution is really helped.

**Comment by bhatnagarm** [comment]
Restarting Docker deamon worked for me.

**Comment by znmeb** [comment]
@NorthNick I've tried this - so far I haven't seen any failures since I disabled fast startup. Can someone who knows the code explain why this works - I don't like "folk theorems" about software.

**Comment by atsaloli** [comment]
I installed Docker Desktop on my Windows 10 laptop yesterday and ran into this problem.  I've restarted Docker for Desktop, which got me going again, and disabled fast startup hoping that will help prevent the problem from happening again.  It would be nice if everything just worked out of the box without having to fiddle with the system settings.  :)  In any case, amazing technology, I appreciate it!

**Comment by orangeswim** [comment]
If Docker Desktop is not set to run automatically on startup, then make sure to start Docker for Windows as administrator. That solved the issue for me. 
I'm on Win 10 with a new installation of Docker Desktop.

**Comment by mshauny** [comment]
Between enabling advanced daemon settings and setting experimental to true worked for me. Running linux containers on a win10. 
![image](https://user-images.githubusercontent.com/3635959/50519738-ff069500-0ac4-11e9-80e0-9b5c80c9582c.png)

**Comment by NorthNick** [comment]
> Between enabling advanced daemon settings and setting experimental to true worked for me. Running linux containers on a win10.

Nice! That worked for me as well. In fact: leaving the Daemon setting as Basic, and just ticking the Experimental features box did the trick on my machine.

**Comment by adaolima** [comment]
In my case I setted a fixed DNS ( 8.8.8.8 ), with unchecked Experimental features , and that worked.

**Comment by rickvanbodegraven** [comment]
I thought we'd already established that, like, at least 100 comments ago. It's about time we get a response from the Docker team themselves how they plan on fixing this.

**Comment by PMT87** [comment]
Why disable fast startup?

Leaving fast startup enabled shouldn't harm anything on your PC — it's a feature built into Windows — but there are a few reasons why you might want to nevertheless disable it.

One of the major reasons is if you're using Wake-on-LAN, which does not work when your PC is shut down with fast startup enabled. Others would like to save the hard drive space that is usually taken up with the hibernation file, and Linux users will likely see complications with dual boot and virtualization. Furthermore, some users have even reported that power buttons on their keyboard will not work and fans in their PCs will continue spinning. These problems are by no means across the board, but they can prove to be frustrating.

Is that really a Docker Problem or more a Windows Problem?

**Comment by slaterbbx** [comment]
So this is still an issue over a year later? Are there any permanent fixes to this? I have this issue too and its super annoying to deal with. I am not launching to production on windows machines ( not sure why anyone would ) but I am working on one, and its crazy that you have to keep doing this and destroying containers that have data in them.

No fix huh?

**Comment by AF8501** [comment]
> I had the experimental features turned off by default. Pressing 'Restart Docker Desktop...' in Reset section fixed it for me on win 10.
> ![image](https://user-images.githubusercontent.com/15012763/49375173-7847dd80-f6fb-11e8-96dd-e23e2da7cdf1.png)

Solved after restarting

**Comment by ProggerPete** [comment]
I'm unsubscribing from this issue because it's clear there is no chance it will ever get fixed.

**Comment by WilliamStam** [comment]
i like how everyone says "restart docker for windows to get it to work" and people still manage to find this thread and post that they found the fix.. restarting docker for windows @AF8501 lol :P 

im testing out docker toolbox at the moment. so far it seems WAY WAY WAY more stable than the native 1 and its faster (not startup tho, just requests etc)

**Comment by gatsbys** [comment]
Seems stupid but stopping all containers and restarting fixed the issue. Nothing more.

**Comment by sawrupesh04** [comment]
I have also faced the same problem in Windows PowerShell. The only way to restart docker.

**Comment by thomas-schuster** [comment]
This was the solution for me as well:
> I had the experimental features turned off by default. Pressing 'Restart Docker Desktop...' in Reset section fixed it for me on win 10.
> ![image](https://user-images.githubusercontent.com/15012763/49375173-7847dd80-f6fb-11e8-96dd-e23e2da7cdf1.png)

Interestingly, even a Windows restart before didn't help! I thought this could "restart" Docker as well, then I found this thread. Thanks!

**Comment by gnubyte** [comment]
Solution is to install Linux on a separate partition and use that for peace of mind.

**Comment by KarenKawaii** [comment]
I have the same error and the experimental features are not checked.
![image](https://user-images.githubusercontent.com/11741384/54889677-6eddaf00-4e6b-11e9-9be8-8678d14b80ba.png)

Another suggestion? please.

**Comment by WilliamStam** [comment]
Just restart docker as almost every post has said.

**Comment by slaterbbx** [comment]
This post is a perfect example of how people do not care to fix things permanently anymore 🤦

I mean, restarting docker fixes NOTHING. What is wrong with people... You restart docker, you realize that it fixes it ( temporarily ) then you come back tomorrow and see it happen again. Then you restart docker again and fix it temporarily again.. This is how you live your life, like a mouse chasing cheese in a maze. Even though people have posted the actual fix on other threads and I have posted a link to that actual fix here.. People keep posting "Oh you just restart it... Worked for me" !!! What?! No it did nothing for you 😂😂 

I just don't get it.

Again... here is a link to the official fix https://github.com/docker/for-win/issues/1038#issuecomment-385889065 read the comment there and realize that the issue is with a problem between docker and windows quick startup. By turning off windows quick startup you will NEVER EVER EVER have this issue again... Do you want to never have this issue again? Then do that! lol

This reminds me of an issue I once saw with a wacom pen and photoshop CC where the pen wouldn't work with Photoshop... Guess what everyone agreed the fix was.. "Pull out the tip of the pen and put it back in" 🤦 Well, I posted the actual fix, it required some code in a setup file for photoshop.... Everyone kept posting "Just pulling the tip of the pen out and putting it back in was the fix for me!! You just have to do it every time." 🤦🤦🤦

Its crazy. If you read this whole thing, then you found the link to the REAL permanent fix. 

If you follow this post for comedy, then lets see how many people after this post will continue to post "just restart docker" lol. Lets get some kind of clap emoji or something to keep a count 😂

Good luck.

**Comment by tbunique** [comment]
@slaterbbx How is disabling a feature considered a REAL fix?

**Comment by slaterbbx** [comment]
> @slaterbbx How is disabling a feature considered a REAL fix?

Well, you could join the docker team and put a fire under their butt to fix the code that is the root issue.

But since they have been ignoring us all for a while now, I am going to guess that wont be happening. 

OR , you can just keep restarting docker. 🤦 

LIke you obviously have been.

**Comment by Jeern** [comment]
@slaterbbx Well you are right. The #restart docker# notifications I have been recieving everyday for more than a year from Github are pretty annoying. But apparently not annoying enough to make me turn of notifications. Says something about my laziness I guess :) The workaround you mention does not work for everyone though ! it is more than a year ago I tried it and it did not help me at all. To be fair these days it happens more rarely to me so perhaps it has had some effect combined with docker updates, windows updates etc. But when I occasionally do get the "Userland proxy thing", guess what I do... Yeah you guessed it... he he :) Since the problem is related to hyper-v I guess it will go away when hyper-v is removed from the eguation someday (heard that on some podcast - is it just a rumour?). Which might also explain why the problem is not prioritized.

**Comment by slaterbbx** [comment]
@Jeern I hope that is not just a rumor. Imo and this is just my opinion, using Hyper-v was such a mistake. But as with all things, I do not know why they did it, but I am sure at the time they had reasons. 

That sucks to hear that the posted "fix" does not solve it for everyone. It solved it for me and so many others but It may have some caveats that are unforeseen and unpredictable as well. ( which could also explain why they avoid this issue. )

I do not know for sure either that the true "root" issue has to do with Hyper-v, but it may be.

Docker toolbox has worked great for me personally. I see no reason for a custom "Windows 10 version" of docker.. Yes it makes setting up Kubernetes super easy for the non techy, but not a good enough reason when even Mac users have to use the long route. 

My personal recommendation is that if you need to run other VMs, use docker toolbox, If you don't need or care about Kubernetes, User docker toolbox and if you need Kubernetes locally and you know how to get stuff working even when it takes more than a button click, save your self a hassle and use docker toolbox 😂

But yea, I just wanted to make that post because everyone was ignoring the fact that the issue had been clearly defined, they kept posting that the "solution" was "X" and I just wanted people to see that there is also a "Y". 

I was not really annoyed lol, just trying to help with a bit of sarcasm and goofing 😂

Like I said, hopefully no one takes offense 🙏

**Comment by WilliamStam** [comment]
we resorted to using linux at work because of docker being a joke on windows. while i appreciate that they brought it to windows.. it has issues.. 

we are a 150 odd staff company that is microsoft bound (office365 everything), but all the devs here are on linux cause of this. really really strange paradigm

**Comment by thomas-schuster** [comment]
@slaterbbx you posted that your link would be an official fix, which it is clearly not! Neither does it help everybody, follow and read contents of your link!

We all agree that a Docker fix would be the perfect solution. I also sense that a straightforward and polite discussion is helpful. Thanks for sharing the other approach. However, your post above would have been helpful if you would just have explained and mentioned the other approach (facts instead of whatever you called sarcasm). Could save some discussions here. I read the other discussion. Seems it does the same thing, besides the fact that the restart is done automatically.

Another thing to discuss would be severity: how often do you have to restart? For me it happened once in 2 years of using Docker in Windows 10 Pro. Thus, I will not consider installing a VM, another system or anything else. However, I acknowledge that you might do so, if that happens much more often. Maybe severity, traceability and complexity are something the Docker team does take into account.

**Comment by msarfrazanwar** [comment]
> Disabling "Experimental Features" in the Settings/Daemon menu, restarting docker and then stopping all images worked for me.
> 
> ![image](https://user-images.githubusercontent.com/7307150/34236256-7976978a-e5ab-11e7-864e-69fca83f5f04.png)

Thanks It worked

**Comment by slaterbbx** [comment]
@thomas-schuster It does not do the same thing, the restart does not happen "automatically" when you turn of "Windows fast startup", can you show me where you found data that infers that by turning off windows fast startup it will automatically restart docker for you?. It simply is the reason it can "sometimes' happen on system wakeup / reboot. That is all.

 I did not start my post with "This should be an official fix, call docker and tell them, I figured it out.." I simply said that restarting docker will not make the issue "go away" indefinitely. ( But I guess some people only experienced this once?, maybe because it has to do with windows fast startup and some hardware handles it differently ) Many already know why this is happening, docker knows why it is happening. Its quite possible the problem is such a complex issue that it is not correctable in its current state and thus this is why there are some future changes in the pipeline according to some "rumors" I guess? ( mentioned by @Jeern )

**This is what I linked too: ( to clarify any confusion )**
1. Go to Control Panel
2. Find Power Options
3. Click on Choose what the power button does in the sidebar on the left.
4. Click on Change settings that are currently unavailable to allow checking / unchecking options under shutdown settings
5. Toggle Turn on fast startup

I have no idea where you read that this causes docker to do a auto restart for you? ( I am guessing some where in that post someone said that? ) But it does not.

Not all... But many people are forced to restart docker on their PC every single time they restart their computer..  If you are one of those people. The options I will post below are your solutions OR you can keep restarting docker every single time ( if you are one of those people / If not, then good for you, honestly, I wish that was my experience also ) but unfortunately, if you use docker professionally and you are having to restart every single time, that is not an option. If you use it as a hobby or just for a few use cases, sure, you can deal with that I am sure.

Above in a humorist manor ( Sorry if that confused anyone ) I mentioned that if you have other reasons, You should use "Docker Toolbox" which yes does use a popular VM created by Oracle, but you are already using a VM known as "Hyper-v" built into window 10 pro. Many professionals will avoid "Hyper-v" unless they are running a legit server or cluster on a dedicated machine. Other than that, on personal / work machines it is very controlling, It is IMPOSSIBLE to run "Hyper-v" with any other VM software, for example "Android studio" Like I mentioned. So in some cases you may have to leave "Docker for Windows", not to mention there are countless other reasons to Use VitualBox over Hyper-v but non of which are related to this topic.

So to clarify, your options are:

1. Restart docker every time it happens. ( unless you where blessed to not have to deal with that, in which case.... What are you doing here? )
2. Turn off a useless feature of windows that offers VERY negligible startup gain ( many people turn it off intentionally to avoid all the quirks that fast startup itself has with some hardware )
3. Use docker toolbox ( My personal recommendation - because hyper-v sucks )
4. Gripe at me for saying these things, Ill take it like a man though 💪 ( oops too playful? )

There ya go, those are the currently available options for this issue. Who knows, maybe if we keep chattering here, they might fix it! Because, at the end of the day, none of us in this topic are actually "contributing" anything, we have not helped solve any issues, we have not made the pull request that fixed the issue, Docker already knows the issue. If I was too playful with this, very sorry that offended your professionalism, that was not my intent, but the bottom line is, this topic is a joke and docker is also treating it like a joke, no official responses in quite some time, no update, no questions, no help, not a thing. Again, I am trying to help, these are the solutions ( at this time ) if you restart docker 1x and it fixes it for you ( 🎉 congrats are in order ) but if you are one of the ones dealing with the daily restart, try turning off "Windows fast startup" or using docker toolbox.

**Comment by thomas-schuster** [comment]
@slaterbbx do not try to fool other people here! This is even more annoying!

Of course you did pretend it would be an official fix. Just in case, I took a screenshot for you:
![image](https://user-images.githubusercontent.com/21131261/55238622-3f48f100-5235-11e9-9d08-df2134e3e384.png)
If you do edit your responses, keep in mind it is github and everybody is able to see previous versions as well.

Regarding automatic restart, I believe that my explanation was a bit imprecise. I read (and meant) that a docker restart is carried out automatically on windows reboot, if fast start-up feature is disabled. I have not tested this, since I do not apply this solution as written above. Sorry for the unclear wording.

@slaterbbx, please note, that I won't discuss this topic any further with you.

**Comment by slaterbbx** [comment]
@thomas-schuster _Your just mad you didn't know the official fix until now._

I **admit** I was being an ass at first, I know how to use GitHub man.

**Exactly why I said** 
_( I was a little douchy in my initial post out of frustration with this issue, sorry, I removed all the douchness I could find )_

**In the other post:**
_( Cleaned up my attitude here as well )_

At least I have the kahunas to admit when I am wrong buddy 💪

Bottom line, its still the **best solution available** _( and the only fix for some )_. So call it what you want.

**Comment by altostec** [comment]
> 
> 
>     1. Stop all the running containers `docker stop $(docker ps -a -q)` then
> 
>     2. Stop the Docker on your machine & `restart` it.

This solution of @raghavmac also worked for me. I found this problem trying to debug a ASP.NET CORE project plus SQLServer 2017 for Linux built toghether with Docker Compose. After reading his solution I realized that an unexpected shutdown happened to my computer and after restart the problem begun. So I restarted the the Docker service (v2.0.0.3 stable) from the context menu of Docker's tray icon and the project ran normally again.

**Comment by EdsionHoang** [comment]
Version 2.0.0.3 (31259)
Channel: stable
Build: 8858db3

This is odd. When I reboot my system, I get the same error.
After I restart docker, it works well. 

> 1. Stop all the running containers `docker stop $(docker ps -a -q)` then
> 
> 2. Stop the Docker on your machine & `restart` it.

This would definitely fix the issue.

**Comment by dalibor983** [comment]
Hello people. This starts to be annoying to restart docer on every windows startup.

I do not wont to disable windows fast startup feature.

If docker cannot restore its state after windows fast startup I think there is a way to detect when windows is shut down with fast startup enabled end stop docker completely like for normal windows shut down.

If this is a windows hyper-v problem lets request from Microsoft to fix the problem.

Or please bring back the old docker toolbox and remove this message from documentation:

> Legacy desktop solution. Docker Toolbox is for older Mac and Windows systems that do not meet the requirements of Docker Desktop for Mac and Docker Desktop for Windows. We recommend updating to the newer applications, if possible.

**Comment by pkjavali** [comment]
> 1. Stop all the running containers `docker stop $(docker ps -a -q)` then
> 2. Stop the Docker on your machine & `restart` it.

This worked for me!

**Comment by dalibor983** [comment]
@pkjavali Do I need to run commands 1. and 2. on every windows startup???

If you have windows fast startup option enabled like this:
![image](https://user-images.githubusercontent.com/226822/56734735-fa48a980-6763-11e9-966e-8ae1271113d0.png)

And you normally shut down machine with:
![image](https://user-images.githubusercontent.com/226822/56734954-82c74a00-6764-11e9-806b-47b4b6881388.png)
(do not restart, you need to shut down the machine normally)
then steps 1. and 2. must be repeated on next windows startup.

**Comment by pkjavali** [comment]
@delibor

> @pkjavali Do I need to run commands 1. and 2. on every windows startup???
> 
> If you have windows fast startup option enabled like this:
> ![image](https://user-images.githubusercontent.com/226822/56734735-fa48a980-6763-11e9-966e-8ae1271113d0.png)
> 
> And you normally shut down machine with:
> ![image](https://user-images.githubusercontent.com/226822/56734954-82c74a00-6764-11e9-806b-47b4b6881388.png)
> (do not restart, you need to shut down the machine normally)
> then steps 1. and 2. must be repeated on next windows startup.

@ dalibor983
I guess you don't need to run each time you start-up. I did it only once and now it works for me every startup.. may be my docker was in a bad state when I got that error. 
I don't know what exactly is your issue? are you getting this error upon each windows start-up? have you set the docker server to start at every boot?
Thanks,

**Comment by dalibor983** [comment]
@pkjavali I do everything like you told and I got this error after regular windows shutdown:
![image](https://user-images.githubusercontent.com/226822/56737261-53b3d700-676a-11e9-999d-a5138c061a23.png)

**Comment by pkjavali** [comment]
@dalibor983,
Sorry, I am getting same error again.. I guess I need to do some more investigation. I'll post if I make any progress.
Thanks,

**Comment by FrankFang** [comment]
That's really annoying!

**Comment by SomeCoolName444** [comment]
I also have this problem. Everyday when Backup team runs backup on this server. Container still runs but you cannot connect to it anymore.

**Comment by tchape** [comment]
solution by @raghavmac works for me, thanks!

**Comment by dweggemans** [comment]
Latest edge version solved it for me

**Comment by tjackadams** [comment]
solution by @raghavmac worked for me.
This was a brand new install and once docker had restarted i got a windows firewall prompt to allow vpnkit access to the network. Not sure why this didn't happen during install but its working now. Thanks

**Comment by FaustoCouto** [comment]
Hello this can look stranger, but for me I check options "Experimental features" and restart then works. Thank you.

**Comment by tbunique** [comment]
> Hello this can look stranger, but for me I check options "Experimental features" and restart then works. Thank you.

It's the restart that makes it work

**Comment by hguven** [comment]
solution by @raghavmac  thanks a lot

**Comment by 0xDefalt** [comment]
Hi guys,
I read all this topic since i was affected by this annoying bug, and disabling experimental features didn't solve the problem. However i found something that maybe could help you.
To reproduce the bug (you can use the mssql docker image for that) just run a new instance, and put your computer on sleep mode then relunch it. You will see that the connection is lost with the container and you cannot restart it : 

```
Error response from daemon: driver failed programming external connectivity on endpoint sqlserver (ac7ddd67e69f8e129c71a5256aa1146cb5d799032de37e09652794c6d8cf2196): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:1433:tcp:172.17.0.2:1433: input/output error
Error: failed to start containers: 1315c
```

I found two ways to "fix" that : Either reboot docker (which can take time), otherwise go the hyper-v console and reboot the Moby Linux VM (which is faster, took me 5 seconds). After that i can start the container again.
Tested the same process on Linux and it doesn't seem to have this bug.

**Comment by itsanjan** [comment]
Its the restart docker that fixed it for me

**Comment by laurencedobson** [comment]
Interestingly, the opposite of what @msarfrazanwar said worked for me - I ***enabled*** "Experimental Features"

**Comment by tbunique** [comment]
@laurencedobson Because you restarted I guess. If you disabled it now, it would do the same.

**Comment by laurencedobson** [comment]
> @laurencedobson Because you restarted I guess. If you disabled it now, it would do the same.

@tbunique I believe you're right. I was going to suggest the same.

**Comment by adamfortuno** [comment]
Windows: 10.0.17763 Build 17763
Docker Desktop Community 2.0.0.3 (31259)
Docker Engine: 18.09.2

Ran into the same issue. I'm running a MongoDB container for development. Container was running fine. I rebooted my laptop. When the host came online, I wasn't able to start the content:

```powershell
PS C:\> docker start c0f
Error response from daemon: driver failed programming external connectivity on endpoint mongodb (1270275ccf2c2b7ba3929e56e710a1a03635863b3c67d65597cae1292e00b466): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:27017:tcp:172.17.0.2:27017: input/output error
Error: failed to start containers: c0f
```
Output from docker events....
```
2019-06-17T10:31:19.690407700-04:00 volume create a48af7127d94585cb2b058f54f1f2bad15becf57a1a4e88066e9baa1cfcafa49 (driver=local)
2019-06-17T10:31:19.706583200-04:00 volume create e86993053d5d10b0d78894e7db7d8472c6aa7d36d35154bc72450d1b7947af18 (driver=local)
2019-06-17T10:31:19.765578600-04:00 container create c0f78550c2d9e45b3107060a18a44dfe6edfa258843b41692bc301b877d573ad (image=mongo, name=mongodb)
2019-06-17T10:31:20.215070800-04:00 network disconnect 466dea49f50a99c1114f222a9c2c7c22c1e7a8cc58ce130982f0b8afa044ae55 (container=c0f78550c2d9e45b3107060a18a44dfe6edfa258843b41692bc301b877d573ad, name=bridge, type=bridge)
2019-06-17T10:31:20.265813100-04:00 volume unmount a48af7127d94585cb2b058f54f1f2bad15becf57a1a4e88066e9baa1cfcafa49 (container=c0f78550c2d9e45b3107060a18a44dfe6edfa258843b41692bc301b877d573ad, driver=local)
2019-06-17T10:31:20.265988100-04:00 volume unmount e86993053d5d10b0d78894e7db7d8472c6aa7d36d35154bc72450d1b7947af18 (container=c0f78550c2d9e45b3107060a18a44dfe6edfa258843b41692bc301b877d573ad, driver=local)
```
I restarted the Docker service, which fixed my problem.

**Comment by bravist** [comment]
> Post updated
> 
> > Desactiving "Experimantal Features" (checked by default) solve my issue

Thanks a lot

**Comment by rikipm** [comment]
Same issue. Restart help, but it not solution

**Comment by dalibor983** [comment]
I think this will not be fixed because docker is planning to switch docker for windows to work with WSL (Windows Subsystem for Linux) instead of Hyper-V (https://devclass.com/2019/06/17/docker-throws-weight-behind-windows-subsystem-for-linux-chucks-hyper-v-option-overboard/).

**Comment by harpsdesire** [comment]
So is there any more persistent solution than restarting Docker constantly or turning off windows quick startup?

**Comment by jasonwilliams** [comment]
> So is there any more persistent solution than restarting Docker constantly or turning off windows quick startup?

Nope, thats why this ticket is so big, as long as its ignored users will come in with the same question. There's also several other tickets in the issue list asking the same thing.

**Comment by yncardoso** [comment]
> 1. Stop all the running containers `docker stop $(docker ps -a -q)` then
> 2. Stop the Docker on your machine & `restart` it.

Just restarting Docker was enough. Thanks.

**Comment by t0rjan** [comment]
Hello everyone, I just encountered this problem. My computer started from hibernation mode. After I restarted Docker Desktop directly, my work started to work.

**Comment by paulsapps** [comment]
Same issue in Linux container mode, right click the tray app and restart docker workaround worked for me.

**Comment by akshaypurohit** [comment]
Although, restarting docker every time is the only solution which seems to be working consistently for everyone. I have observed that restarting "Host Network Service" Windows service also works and is faster way to workaround this problem.

**Comment by niclarcipretti** [comment]
@akshaypurohit 's solution didn't work for me.

Imho, restart Docker is not a viable solution (once Docker take centuries to restart on windows).
I do understand that docker will evolve to let hyper-v finally go, but, I do think this issue should be addressed once some production environments will not be able to update Docker until it's stable or because of some company policy.

Can any Docker engineer give some thoughts on this issue?

Thanks in advance.

Nikao

**Comment by guiprerod** [comment]
In my case it was working when I selected "Experimental features"

**Comment by niclarcipretti** [comment]
@guiprerod, read the entire thread plz.

**Comment by judos** [comment]
@akshaypurohit @harpsdesire

**Solution**: Simply disable "Start Docker Desktop when you log in".
But keep it at start-up program (e.g. copy a link to docker into your startup folder here:
%USERPROFILE%\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup

This will make Docker start later after logging in and that works for many people.

**Comment by jasonwilliams** [comment]
This issue should be fixed by https://github.com/docker/for-win/issues/2722#issuecomment-514530476

**Comment by rikipm** [comment]
I stop getting this problem after last update

**Comment by rikipm** [comment]
It works, I stop getting this problem after last update.

**Comment by nickmafra** [comment]
I was with the message

`docker run -d -e ALLOW_OVERRIDE=true -v /:/app/exchange -p 81:81 --name exchange nickmafra/exchange
docker: Error response from daemon: Conflict. The container name "/exchange" is already in use by container "2650193e659df17ebf03f7f4b730670255049da3a86470859a8538eda11ebe27". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.`

I changed ports and it fixed the issue

`docker run -d -e ALLOW_OVERRIDE=true -v /:/app/exchange -p 81:81 --name exchange nickmafra/exchange`

**Comment by nickmafra** [comment]
I was with the following message:

`docker run -d -e ALLOW_OVERRIDE=true -v /:/app/exchange -p 80:80 --name exchange nickmafra/exchange
2650193e659df17ebf03f7f4b730670255049da3a86470859a8538eda11ebe27
docker: Error response from daemon: driver failed programming external connectivity on endpoint exchange (23e2644665fff7f69074b64cd3aec640121780cb1519262230ccb71ed4bf533d): Error starting userland proxy: /forwards/expose/port returned unexpected status: 500.`

I changed ports for 81 and it fixed the issue

**Comment by jakubsgil** [comment]
Hi there, I have upgraded to the newest docker version (2.1.0.1) and then started to see this issue. Restarting docker does not seem to work/resolve this. Only see this problem go away is when I reinstall docker. Is there some sort of port or firewall related issue that I might be seeing from an enterprise IT restriction?

**Comment by Vespira** [comment]
> 
> 
> Hi there, I have upgraded to the newest docker version (2.1.0.1) and then started to see this issue. Restarting docker does not seem to work/resolve this. Only see this problem go away is when I reinstall docker. Is there some sort of port or firewall related issue that I might be seeing from an enterprise IT restriction?

Hi, I posted a similary message and I just want you to know something. A wild guess make me think that when you shutdown your computer while docker is running some containers could corrupt the state of networks or containers (for me it was networks). I resetted to factory settings Docker, but next time I'll try only to remove concerned docker network, and try to run again my docker-compose deploy to see if recreating network is sufficent.

Hope it helps

**Comment by andrewfinnell** [comment]
Our team also experiences these issues. I'm wondering if @Vespira is on to something regarding running Containers and the machine going to sleep or being shut down. Either way, this is quite unfortunate that it's been years and this still occurs.

```
Successfully built 12bd2d95a54d
Successfully tagged *************:latest
Creating eureka-1 ...
Creating ************* ...
Creating *************  ... error

Creating eureka-1            ... error
f7105d8a7e7e3948f1f25ff10a5d32d246ae68ebb67edcf170ffde000): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:8080:tcp:172.20.0.2:8080: input/output error

Creating ************* ... error
17eb30a47aff4b854f3370083dd): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:8761:tcp:172.20.0.3:8761: input/output error

ERROR: for *************  Cannot start service *************: driver failed programming external connectivity on endpoint *************(c993dd36af5da2482f7d48cbc884002ce4cb3fc219cf550b017d1f13eabac2a1): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:5432:tcp:172.20.0.4:5432: input/output error

ERROR: for *************  Cannot start service *************: driver failed programming external connectivity on endpoint ************* (9864f9bf7105d8a7e7e3948f1f25ff10a5d32d246ae68ebb67edcf170ffde000): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:8080:tcp:172.20.0.2:8080: input/output error

ERROR: for eureka-1  Cannot start service eureka-1: driver failed programming external connectivity on endpoint eureka-1 (9588322c5bf824f0d40570802181e3e2d647017eb30a47aff4b854f3370083dd): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:8761:tcp:172.20.0.3:8761: input/output error

ERROR: for *************  Cannot start service *************: driver failed programming external connectivity on endpoint ************* (c993dd36af5da2482f7d48cbc884002ce4cb3fc219cf550b017d1f13eabac2a1): Error starting userland proxy: mkdir /port/tcp:0.0.0.0:5432:tcp:172.20.0.4:5432: input/output error
ERROR: Encountered errors while bringing up the project.
```

**Comment by mreilaender** [comment]
I think for each port specified to be exposed for a container docker starts a process, which is not correctly terminated, when docker or the container is shut down.

The solution I came up with:

1. Stop docker
2. Find the process(es) which is using the port(s) and kill them
3. Start docker
4. Start your containers

Powershell command to kill a process which is listening on a specified port (requires admin privileges):
```
Stop-Process -Id (Get-Process -Id (Get-NetTCPConnection -LocalPort <portNumber>).OwningProcess).Id
```
Substitude <portNumber> with the actual port number

**Comment by dobrik** [comment]
1. Kill process **com.docker.backend.exe** 
2. Restart docker desktop

**Comment by nguyenhongphu** [comment]
If anyone using Windows, a simple solution would be to install whatever pending updates of Windows, then restart your computer! It worked for me :)

**Comment by omarmahili** [comment]
"Reset to factory defaults..." solved the issue. 
**This action will remove ALL docker containers and images!**

![docker_reset](https://user-images.githubusercontent.com/3227034/68375997-192eec00-0148-11ea-89de-2c6bbdddb184.png)

**Comment by Vespira** [comment]
> 
> 
> "Reset to factory defaults..." solved the issue.
> **This action will remove ALL docker containers and images!**
> 
> ![docker_reset](https://user-images.githubusercontent.com/3227034/68375997-192eec00-0148-11ea-89de-2c6bbdddb184.png)

Yes it was mentionned few times here and in other threads about this... But for most of us, it can't be a viable solution because re-downloading tons of images and recreating test data is damn slow.

My current tip for preventing this was also mentionned; you can disable fastboot on Windows, in the Power options to prevent Docker from being corrupted by an hybrid shutdown ;)

**Comment by igorpopryadukhin** [comment]
Do not let Docker start with Windows

**Comment by marcinpejski** [comment]
> 
> 
>     1. Kill process **com.docker.backend.exe**
> 
>     2. Restart docker desktop

Killing all the processess of docker (backend/proxy/service) and quitting and starting docker once again did the trick for me.

**Comment by AlexQuidditch** [comment]
I found solution for Windows 10:
Win + R (or cmd)
0. Stop Docker Desktop
1. $ services.msc
2. Find "Docker Desktop Service"
3. Restart it
4. Run Docker Desktop

**Comment by spawnrider** [comment]
> I found solution for Windows 10:
> 
> 1. Stop Docker Desktop
> 2. Win + R (or cmd) > `$ services.msc`
> 3. Find "Docker Desktop Service"
> 4. Restart it
> 5. Run Docker Desktop

This works for me on Windows 10 and Docker Desktop 2.1.0.5

**Comment by elonderin** [comment]
* problem happened to me too
* i think the culprit was, that i changed CPU and RAM in the advanced settings, which caused the docker desktop to restart while containers were running with `--restart unless-stopped'
* Windows 10 and Docker Desktop 2.1.0.5
* after trying the advice to shutdown and restart this wouldn't work at first (tried it 2x b/c i had not stopped docker desktop before)
* still, after doing the exact steps from above it failed
* then i opened the settings and went back to advanced settings , where i noticed that they had revered to the default settings 
* however, i dont remember having reset them 
* after that it suddenly worked !?

thoughts:
* could it be that the restarting the service actually takes longer than what the windows status says ?
